era/threat/report.js

```diff
@@ -0,0 +1,81 @@
+import { fetchWCLv1 } from "./wcl.js";
+import { Fight } from "./fight.js";
+
+export class Report {
+  /**
+   * @param {string} reportId
+   * @param {import("../base.js").GameVersionConfig} config
+   */
+  constructor(config, reportId) {
+    /**
+     * @type {import("../base.js").GameVersionConfig}
+     */
+    this.config = config;
+    /**
+     * @type {string}
+     */
+    this.reportId = reportId;
+    /**
+     * @type {Record<string, Fight>}
+     */
+    this.fights = {};
+
+    this.data = undefined;
+  }
+
+  async fetch() {
+    if (this.data) return;
```

Comment:
Minor change here from `if ('data' in this) return;`, which broke type checking.